src/ert/shared/cli/model_factory.py

```diff
@@ -8,21 +8,23 @@
 from ert.shared.models.single_test_run import SingleTestRun
 
 
-def create_model(ert, ensemble_size, current_case_name, args):
+def create_model(ert, ensemble_size, current_case_name, args, id_):
     # Setup model
     if args.mode == "test_run":
         model = _setup_single_test_run(ert)
```

Comment:
Is it intentional to not associate test-runs with the experiment server?

Reply:
I see, it is defaulted to "0" in the    `SingleTestRun.__init__`

Reply:
This leads to a crash when rebased on top of `experiment_api`, and then tested with `test_run` and with experiment-server applied. Seemingly resolved by passing the exp_id to `_setup_single_test_run`.

Reply:
Yeh, getting a `KeyError` when asking the `experiment_server` to run experiment with given id, as it ignores the id created in the `async_run`. I've made it propagate, not sure if I like how it's done though, as the `SingleTestRun` doesn't include the `queue_config` - hence instantiating the object suddenly differs from the rest.. (the `id` is the third argument instead of the fourth...).

Thanks for picking up on it though!